docs/map.rst

```diff
@@ -0,0 +1,602 @@
+
+There's another incredibly useful function for constructing a map from a list::
+
+    Map.fromListWith :: Ord k => (a -> a -> a) -> [(k, a)] -> Map.Map k a
+    Map.fromListWith f xs = ...
+
+:haddock_short:`Data.Map.Strict#fromListWith` allows you to build a map from a
+list ``xs`` with repeated keys, where ``f`` is used to "combine" (or "choose")
+values with the same key.
+
+::
+
+    -- Build a map from a list, but only keep the largest value for each key.
+    Map.fromListWith max [("a", 2), ("a", 1), ("b", 2)]
+    > fromList [("a",2),("b",2)]
+
+    -- Build a histogram from a list of elements.
+    Map.fromListWith (+) (map (\x -> (x, 1)) ["a", "a", "b", "c", "c", "c"])
+    > fromList [("a",2),("b",1),("c",3)]
+
+    -- Build a map from a list, combining the string values for the same key.
+    Map.fromListWith (++) [(1, "a"), (1, "b"), (2, "x"), (2, "y")]
+    > fromList [(1,"ba"),(2,"yx")]
+
+
+
+Create a list from a map
+""""""""""""""""""""""""
+
+::
+
+    Map.elems :: Map k v -> [v]
+    Map.elems m = ...
+
+:haddock_short:`Data.Map.Strict#elems` returns a list of values held in the map
+``m``.
+
+::
+
+    Map.toAscList, Map.toList, Map.assocs :: Map k v -> [(k, v)]
+    Map.toAscList m = ...
+
+.. NOTE::
+   These all do the same thing; use ``toAscList`` because its name indicates
+   the ordering.
+
+.. NOTE::
+   ``Map.toList`` is **not** the same as ``Foldable.toList``; the latter is
+   equivalent to ``elems``.
+
+:haddock_short:`Data.Map.Strict#toAscList`,
+:haddock_short:`Data.Map.Strict#toList`, and 
+:haddock_short:`Data.Map.Strict#assocs` returns a list containing the (key,
+value) pairs in the map ``m`` in *ascending* key order.
+
+::
+
+    Map.toDescList :: Map k v -> [(k, v)]
+    Map.toDescList m = ...
+
+:haddock_short:`Data.Map.Strict#toDescList` returns a list containing the (key,
+value) pairs in the map ``m`` in *descending* key order.
+
+::
+
+    Map.elems (Map.fromList [(1,"one"), (2,"two"), (3,"three")])
+    > ["one","two","three"]
+
+    Map.toAscList (Map.fromList [(1,"one"), (2,"two"), (3,"three")])
+    > [(1,"one"),(2,"two"),(3,"three")]
+
+    Map.toDescList (Map.fromList [(1,"one"), (2,"two"), (3,"three")])
+    > [(3,"three"),(2,"two"),(1,"one")]
+
+
+Querying
+^^^^^^^^
+
+Check if a map is empty
+"""""""""""""""""""""""
+
+::
+
+    Map.null :: Map k v -> Bool
+    Map.null m = ...
+
+:haddock_short:`Data.Map.Strict#null` returns ``True`` if the map ``m`` is
+empty, ``False`` otherwise.
+
+::
+
+    Map.null Map.empty
+    > True
+
+    Map.null (Map.fromList [(1,"one")])
+    > False
+
+The number of entries in a map
+""""""""""""""""""""""""""""""
+
+::
+
+    Map.size :: Map k v -> Int
+    Map.size m = ...
+
+:haddock_short:`Data.Map.Strict#size` returns the number of entries in the map
+``m``.
+
+::
+
+    Map.size Map.empty
+    > 0
+
+    Map.size (Map.fromList [(1,"one"), (2,"two"), (3,"three")])
+    > 3
+
+Lookup an entry in the map (lookup)
```

Comment:
s/Lookup an entry/Look up an entry

Reply:
Also, please float `lookup` to the top of the query section. It's the most important query by far.

Reply:
Done.